### PR TITLE
Add pod state on warning events

### PR DESCRIPTION
**Pull request by nilsgstrabo**

### CODE REVIEW

File: api/events/event_handler.go [hunk 1]
```diff
@@ -45,11 +46,51 @@ func (eh *eventHandler) getEvents(namespace string) ([]*eventModels.Event, error
 		return nil, err
 	}
 
+	//objectGetter := objectGetter{kubeClient: eh.kubeClient}
```
Comment (olemoi) [inline api/events/event_handler.go:49]: Remove this?

Comment (nilsgstrabo) [reply api/events/event_handler.go]: Fixed.

File: api/events/event_handler_test.go [hunk 2]
```diff
@@ -49,13 +47,70 @@ func Test_EventHandler_GetEventsForRadixApplication(t *testing.T) {
...
 }
 
-func createKubernetesEvent(client *kubefake.Clientset, ns, name, involvedObjectName string) {
-	client.CoreV1().Events(ns).CreateWithEventNamespace(&v1.Event{
+func Test_EventHandler_GetEvents_PodState(t *testing.T) {
+	appName, envName := "app", "env"
+	appNamespace := k8sObjectUtils.GetEnvironmentNamespace(appName, envName)
+
+	t.Run("ObjectState is nil for normal event type", func(t *testing.T) {
+		kubeClient := kubefake.NewSimpleClientset()
```
Comment (olemoi) [inline api/events/event_handler_test.go:55]: Extract this to a common initializer?

File: api/events/event_handler_test.go [hunk 3]
```diff
@@ -49,13 +47,70 @@ func Test_EventHandler_GetEventsForRadixApplication(t *testing.T) {
...
+		createKubernetesPod(kubeClient, "pod1", appNamespace, true, true, 0)
+		ra := builders.NewRadixApplicationBuilder().WithAppName(appName).BuildRA()
+		eventHandler := Init(kubeClient)
+		events, _ := eventHandler.GetEvents(RadixEnvironmentNamespace(ra, envName))
+		assert.Len(t, events, 1)
+		assert.Nil(t, events[0].InvolvedObjectState)
+	})
+
+	t.Run("ObjectState has Pod state for warning event type", func(t *testing.T) {
+		kubeClient := kubefake.NewSimpleClientset()
```
Comment (olemoi) [inline api/events/event_handler_test.go:66]: Extract this to a common initializer?

File: api/events/event_handler_test.go [hunk 4]
```diff
@@ -49,13 +47,70 @@ func Test_EventHandler_GetEventsForRadixApplication(t *testing.T) {
...
+		ra := builders.NewRadixApplicationBuilder().WithAppName(appName).BuildRA()
+		eventHandler := Init(kubeClient)
+		events, _ := eventHandler.GetEvents(RadixEnvironmentNamespace(ra, envName))
+		assert.Len(t, events, 1)
+		assert.NotNil(t, events[0].InvolvedObjectState)
+		assert.NotNil(t, events[0].InvolvedObjectState.Pod)
+	})
+
+	t.Run("ObjectState is nil for warning event type when pod not exist", func(t *testing.T) {
+		kubeClient := kubefake.NewSimpleClientset()
```
Comment (olemoi) [inline api/events/event_handler_test.go:78]: Extract this to a common initializer?

File: api/events/event_handler_test.go [hunk 5]
```diff
@@ -49,13 +47,70 @@ func Test_EventHandler_GetEventsForRadixApplication(t *testing.T) {
...
+		createKubernetesEvent(kubeClient, appNamespace, "ev1", "Normal", "pod1", "Pod")
+		ra := builders.NewRadixApplicationBuilder().WithAppName(appName).BuildRA()
```
Comment (olemoi) [inline api/events/event_handler_test.go:80]: Could also include ra in initializer if appname is not variable

File: api/events/event_handler_test.go (hunk 4 above)
Comment (nilsgstrabo) [reply api/events/event_handler_test.go:78]: Each test need it's own fake kubeclient with k8s objects.

**Review by olemoi — APPROVED** [review]

File: api/events/event_handler_test.go (hunk 5 above)
Comment (nilsgstrabo) [reply api/events/event_handler_test.go]: Agree. Fixed.

**Pull request merged**